pallets/creditcoin/src/ocw/tasks/collect_coins.rs

```diff
@@ -126,14 +126,19 @@ impl<T: CreditcoinConfig> Pallet<T> {
 			to,
 			tx_id,
 			contract: DeployedContract { address, chain },
-			contract_type: _,
+			contract_type,
 		} = u_cc;
 		let rpc_url = &chain.rpc_url()?;
 		let tx = ocw::eth_get_transaction(tx_id, rpc_url)?;
 		let tx_receipt = rpc::eth_get_transaction_receipt(tx_id, rpc_url)?;
 		let eth_tip = rpc::eth_get_block_number(rpc_url)?;
 
-		let amount = validate_collect_coins(to, &tx_receipt, &tx, eth_tip, address)?;
+		let mut amount = validate_collect_coins(to, &tx_receipt, &tx, eth_tip, address)?;
+
+		// GATE -> CTC is swapped 2:1
+		if *contract_type == ContractType::GATE {
+			amount = amount.div(sp_core::U256::from_dec_str("2").unwrap());
```

Comment:
```suggestion
			amount /= 2;
```


Sort of a side note, but in general, unwrap or other functions that can panic are a code smell in on-chain code. It's correct in this case, as it can truly never fail, but you have to be careful to show that it will _never_ panic.